src/lighteval/models/openai_model.py

```diff
@@ -0,0 +1,241 @@
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import os
+import time
+from concurrent.futures import ThreadPoolExecutor
+from typing import Optional
+
+from tqdm import tqdm
+
+from lighteval.data import GenerativeTaskDataset, LoglikelihoodDataset
+from lighteval.logging.hierarchical_logger import hlog_warn
+from lighteval.models.abstract_model import LightevalModel
+from lighteval.models.endpoint_model import ModelInfo
+from lighteval.models.model_output import (
+    GenerativeResponse,
+    LoglikelihoodResponse,
+    LoglikelihoodSingleTokenResponse,
+)
+from lighteval.tasks.requests import (
+    GreedyUntilRequest,
+    LoglikelihoodRequest,
+    LoglikelihoodRollingRequest,
+    LoglikelihoodSingleTokenRequest,
+)
+from lighteval.utils.imports import is_openai_available
+
+
+if is_openai_available():
+    import logging
+
+    import tiktoken
+    from openai import OpenAI
+
+    logging.getLogger("openai").setLevel(logging.ERROR)
+    logging.getLogger("httpx").setLevel(logging.ERROR)
+
+
+class OpenAIClient(LightevalModel):
+    _DEFAULT_MAX_LENGTH: int = 4096
+
+    def __init__(self, config, env_config) -> None:
+        api_key = os.environ["OPENAI_API_KEY"]
+        self.client = OpenAI(api_key=api_key)
+
+        self.model_info = ModelInfo(
+            model_name=config.model,
+            model_sha="",
+            model_dtype=None,
+            model_size="",
+        )
+        self.API_MAX_RETRY = 5
+        self.API_RETRY_SLEEP = 3
+        self.CONCURENT_CALLS = 100
+        self.model = config.model
+        self._tokenizer = tiktoken.encoding_for_model(self.model)
+        self.pairwise_tokenization = False
+
+    def __call_api(self, prompt, return_logits, max_new_tokens, num_samples, logit_bias):
+        for _ in range(self.API_MAX_RETRY):
+            try:
+                response = self.client.chat.completions.create(
+                    model=self.model,
+                    messages=[{"role": "user", "content": prompt}],
+                    response_format={"type": "text"},
+                    max_tokens=max_new_tokens if max_new_tokens > 0 else 100,
+                    logprobs=return_logits,
+                    top_logprobs=3,
+                    logit_bias=logit_bias,
+                    n=num_samples,
+                )
+                return response
+            except Exception as e:
+                hlog_warn(f"{type(e), e}")
+                time.sleep(self.API_RETRY_SLEEP)
+        raise Exception("Failed to get response from the API")
+
+    def __call_api_parallel(self, prompts, return_logits, max_new_tokens, num_samples, logit_bias=None):
+        results = []
+        return_logitss = [return_logits for _ in prompts] if not isinstance(return_logits, list) else return_logits
```

Comment:
Could be worth creating a function to extend a value as a list if not a list.
Are you sure that if return_logits is a list, it will be the same size as prompts?

Reply:
Mhh not sure this needs a function, I we only need it here and it would add complexity to the code for no reason.

Actually, all those args should never be lists expect for `logit_bias`

Reply:
Then we'd need to check this :)

Reply:
Should we ? If they are of different size, the ThreadPoolExecutor is just going to fail immediatly

Reply:
Will the return message be clear?

Reply:
It's not very clear but just added an assert